org.eclipse.jdt.core.manipulation/refactoring/org/eclipse/jdt/internal/corext/refactoring/sef/SelfEncapsulateFieldCompositeRefactoring.java

```diff
@@ -0,0 +1,229 @@
+
+import org.eclipse.jdt.internal.core.manipulation.JavaManipulationPlugin;
+import org.eclipse.jdt.internal.core.manipulation.StubUtility;
+import org.eclipse.jdt.internal.core.manipulation.util.BasicElementLabels;
+import org.eclipse.jdt.internal.core.refactoring.descriptors.RefactoringSignatureDescriptorFactory;
+import org.eclipse.jdt.internal.corext.dom.IASTSharedValues;
+import org.eclipse.jdt.internal.corext.refactoring.RefactoringAvailabilityTesterCore;
+import org.eclipse.jdt.internal.corext.refactoring.RefactoringCoreMessages;
+import org.eclipse.jdt.internal.corext.refactoring.changes.DynamicValidationRefactoringChange;
+import org.eclipse.jdt.internal.corext.refactoring.util.RefactoringASTParser;
+import org.eclipse.jdt.internal.corext.refactoring.util.TextChangeManager;
+import org.eclipse.jdt.internal.corext.util.Messages;
+
+public class SelfEncapsulateFieldCompositeRefactoring extends Refactoring {
+
+	private final List<SelfEncapsulateFieldRefactoring> fRefactorings;
+	private final HashMap<ICompilationUnit, ImportRewrite> fImportRewritesMap = new HashMap<>();
+	private final HashMap<ICompilationUnit, ASTRewrite> fRewritersMap = new HashMap<>();
+	private final HashMap<ICompilationUnit, CompilationUnit> fRootsMap = new HashMap<>();
+	private final HashMap<ICompilationUnit, List<TextEditGroup>> fDescriptionsMap = new HashMap<>();
+	private final List<TextChange> changes = new ArrayList<>();
+
+	private static final String NO_NAME= ""; //$NON-NLS-1$
+
+	public List<SelfEncapsulateFieldRefactoring> getRefactorings() {
+		return fRefactorings;
+	}
+
+	public SelfEncapsulateFieldCompositeRefactoring(List<IField> fields) throws JavaModelException {
+		fRefactorings = getRefactoringFromFields(fields);
+	}
+
+	public List<SelfEncapsulateFieldRefactoring> getRefactoringFromFields(List<IField> fields) throws JavaModelException {
+		List<SelfEncapsulateFieldRefactoring> refactorings = new ArrayList<>();
+		for(IField field : fields) {
+			if (RefactoringAvailabilityTesterCore.isSelfEncapsulateAvailable(field)) {
+				refactorings.add(new SelfEncapsulateFieldRefactoring(field, this));
+			}
+		}
+		return refactorings;
+	}
+
+	@Override
+	public String getName() {
+		return RefactoringCoreMessages.SelfEncapsulateField_name;
+	}
+
+	@Override
+	public RefactoringStatus checkInitialConditions(IProgressMonitor pm) throws CoreException, OperationCanceledException {
+
+		RefactoringStatus fInitialConditions = new RefactoringStatus();
+		for(SelfEncapsulateFieldRefactoring selfEncapsulateFieldRefactoring : fRefactorings) {
+			fInitialConditions.merge(selfEncapsulateFieldRefactoring.checkInitialConditions(pm));
+		}
+		return fInitialConditions;
+	}
+
+	@Override
+	public Change createChange(IProgressMonitor pm) throws CoreException, OperationCanceledException {
+		pm.beginTask(NO_NAME, changes.size());
+		pm.setTaskName(RefactoringCoreMessages.SelfEncapsulateField_create_changes);
```

Comment:
Actually, I meant to remove the setTaskName().  According to IProgressMonitor docs, normally there is no need to call this method.  If you want the SelfEncapsulateField_create_changes message, just specify it in the beginTask since you never change it nor pass the pm anywhere.

Reply:
Done